src/enclave/enclavetypes.h

```diff
@@ -11,69 +11,127 @@ namespace enclave
-    // Request payload specific attributes
-    struct request
-    {
-      uint64_t seq_no;
-    };
-    struct request req;
-
-    bool is_create_request = false;
+    size_t client_session_id = InvalidSessionId;
+    std::vector<uint8_t> caller_cert = {};
 
     //
     // Only set in the case of a forwarded RPC
     //
-    struct forwarded
+    struct Forwarded
     {
       // Initialised when forwarded context is created
       const size_t client_session_id;
       const ccf::CallerId caller_id;
 
-      forwarded(size_t client_session_id_, ccf::CallerId caller_id_) :
+      Forwarded(size_t client_session_id_, ccf::CallerId caller_id_) :
         client_session_id(client_session_id_),
         caller_id(caller_id_)
       {}
     };
-    std::optional<struct forwarded> fwd = std::nullopt;
+    std::optional<Forwarded> fwd = std::nullopt;
 
     // Constructor used for non-forwarded RPC
-    RPCContext(
-      size_t client_session_id_,
-      const std::vector<uint8_t>& caller_cert_,
-      ccf::ActorsType actor_ = ccf::ActorsType::unknown) :
+    SessionContext(
+      size_t client_session_id_, const std::vector<uint8_t>& caller_cert_) :
       client_session_id(client_session_id_),
-      caller_cert(caller_cert_),
-      actor(actor_)
+      caller_cert(caller_cert_)
     {}
 
     // Constructor used for forwarded and PBFT RPC
-    RPCContext(
+    SessionContext(
       size_t fwd_session_id_,
       ccf::CallerId caller_id_,
-      ccf::ActorsType actor_ = ccf::ActorsType::unknown,
       const std::vector<uint8_t>& caller_cert_ = {}) :
-      fwd(std::make_optional<struct forwarded>(fwd_session_id_, caller_id_)),
-      actor(actor_),
+      fwd(std::make_optional<Forwarded>(fwd_session_id_, caller_id_)),
       caller_cert(caller_cert_)
     {}
   };
 
+  struct RPCContext
+  {
+    SessionContext session;
+
+    // Packing format of original request, should be used to pack response
+    std::optional<jsonrpc::Pack> pack = std::nullopt;
+
+    // TODO: Avoid unnecessary copies
+    std::vector<uint8_t> raw = {};
+
+    nlohmann::json unpacked_rpc = {};
+
+    std::optional<ccf::SignedReq> signed_request = std::nullopt;
+
+    // Actor type to dispatch to appropriate frontend
+    ccf::ActorsType actor = ccf::ActorsType::unknown;
+
+    // Method indicates specific handler for this request
+    std::string method = {};
+
+    uint64_t seq_no = {};
+
+    nlohmann::json params = {};
+
+    bool is_create_request = false;
+
+    RPCContext(const SessionContext& s) : session(s) {}
+  };
+
+  inline void parse_rpc_context(RPCContext& rpc_ctx, const nlohmann::json& rpc)
+  {
+    const auto sig_it = rpc.find(jsonrpc::SIG);
+    if (sig_it != rpc.end())
+    {
+      rpc_ctx.unpacked_rpc = rpc.at(jsonrpc::REQ);
+      ccf::SignedReq signed_req;
+      signed_req.sig = sig_it->get<decltype(signed_req.sig)>();
+      signed_req.req = nlohmann::json::to_msgpack(rpc_ctx.unpacked_rpc);
```

Comment:
It occurs to me that once we stop this madness, the client sig tx perf test will surely get better.